severn/api/requirements.py

```diff
@@ -0,0 +1,282 @@
+            raise ValueError("invalid version string")
+
+        version = match.groupdict()
+        raw_release = version["release"]
+
+        if "*" in raw_release:
+            version["comparitor"] = "~="
+            raw_release = raw_release.replace("*", "0")
+
+        release = [int(v) for v in raw_release.split(".")]
+        n_parts = len(release)
+        release += [0] * (3 - n_parts)
+
+        def maybe_int(value: Optional[str]) -> Optional[int]:
+            return int(value) if value else None
+
+        kwargs = (
+            {version["pre_l"]: maybe_int(version["pre_n"])} if version["pre"] else {}
+        )
+
+        return cls(
+            version["comparitor"],
+            epoch=maybe_int(version["epoch"]),
+            major=release[0],
+            minor=release[1],
+            patch=release[2],
+            post1=maybe_int(version["post_n1"]),
+            post2=maybe_int(version["post_n2"]),
+            dev=maybe_int(version["dev_n"]),
+            release_specificity=n_parts,
+            **kwargs,
+        )
+
+    def _tilde_callback(
+        self,
+        constraint: Tuple[Union[int, float], ...],
+        requirement: Tuple[Union[int, float], ...],
+    ) -> bool:
+        rs = self._release_specificity
+        return constraint[:rs] == requirement[:rs] and requirement[rs] >= constraint[rs]
+
+    def likes_version(self, version: str, /) -> bool:
+        other = Constraint.from_string(f"=={version}")
+        COMPARITOR_MAPPING = {
+            "==": self.as_tuple.__eq__,
+            "!=": self.as_tuple.__ne__,
+            ">": self.as_tuple.__lt__,
+            ">=": self.as_tuple.__le__,
+            "<": self.as_tuple.__gt__,
+            "<=": self.as_tuple.__ge__,
+            "~=": partial(self._tilde_callback, self.as_tuple),
+        }
+        return COMPARITOR_MAPPING[self.comparitor](other.as_tuple)
+
+
+class Dependency:
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        *,
+        constraints: Optional[List[str]] = None,
+        env_markers: Optional[List[str]] = None,
+        extras: Optional[List[str]] = None,
+        location: Optional[Union[str, Path]] = None,
+        editable: bool = False,
+    ) -> None:
+        self.name = name
+        self.constraints = (
+            [Constraint.from_string(c) for c in constraints] if constraints else None
+        )
+        self.env_markers = env_markers  # TODO: Convert to a dict.
+        self.extras = extras
+        self.location = location
+        self.editable = editable
+
+    def likes_version(self, version: str) -> bool:
+        return (
+            all(c.likes_version(version) for c in self.constraints)
+            if self.constraints
+            else True
+        )
+
+
+class RequirementsFile:
+    def __init__(self, path: Union[str, Path]) -> None:
+        self.path = path if isinstance(path, Path) else Path(path)
+
+    def parse(self) -> List[Dependency]:
```

Comment:
May be good to either throw errors or warnings if it finds something it can't handle? Like editable installs currently.